shell_integration/dolphin/ownclouddolphinpluginhelper.cpp

```diff
@@ -98,47 +98,57 @@ void OwncloudDolphinPluginHelper::slotReadyRead()
 {
     while (_socket.bytesAvailable()) {
         _line += _socket.readLine();
-        if (!_line.endsWith("\n"))
+        if (!_line.endsWith("\n")) {
             continue;
+        }
         QByteArray line;
         qSwap(line, _line);
         line.chop(1);
         if (line.isEmpty())
             continue;
+        int firstColon = line.indexOf(':');
+        if (firstColon == -1) {
+            continue;
+        }
+        // get the command (at begin of line, before first ':')
+        const QByteArray command = line.left(firstColon);
+        // rest of line contains the information
+        QByteArray info = line.right(line.size() - firstColon - 1);
```

Comment:
Might be me but I find  `line.mid(firstColon + 1)` more intuitive.
Also const.
If you use midRef it event removes the need to create a copy of the string.

Reply:
Yes, line.mid() IS more intuitive